### PR TITLE
Auth/PM-34506 - LoginStrategyService - Refactor cache and timeout out into own services

### PR DESCRIPTION
**Pull request by JaredSnider-Bitwarden**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-34506
https://bitwarden.atlassian.net/browse/PM-34867

## 📔 Objective

To refactor the `LoginStrategyService` and extract the login cache and session timeout domains out into their own single responsibility services. 

## Detailed Changes

<details>
<summary>Abstractions</summary>

**`libs/auth/src/common/abstractions/login-strategy-cache.service.ts`** *(new)*
1. Defines `LoginStrategyCacheService` abstract class exposing three observable streams (`currentAuthType$`, `cacheData$`, `cacheExpiration$`) and four write methods (`setCurrentAuthType`, `setCacheData`, `setCacheExpiration`, `clearCache`) — isolates all mid-auth cache state behind a dedicated contract so `LoginStrategyService` no longer owns raw `GlobalState` references.

**`libs/auth/src/common/abstractions/login-strategy-session-timeout.service.ts`** *(new)*
1. Defines `LoginStrategySessionTimeoutService` abstract class with `loginSessionTimeout$` (emits on expiry), `startSessionTimeout()`, and `cancelSessionTimeout()` — moves the session-lifetime contract out of `LoginStrategyServiceAbstraction` and into its own purpose-built interface.

**`libs/auth/src/common/abstractions/index.ts`**
1. Added re-exports for both new abstractions so consumers can import from the `@bitwarden/auth/common` barrel.

**`libs/auth/src/common/abstractions/login-strategy.service.ts`**
1. Removed `authenticationSessionTimeout$: Observable<boolean>` — the observable is now owned by `LoginStrategySessionTimeoutService`, eliminating a conceptual mismatch where the login-strategy abstraction exposed timeout lifecycle state.

</details>

<details>
<summary>Services</summary>

**`libs/auth/src/common/services/login-strategies/login-strategy-cache.service.ts`** *(new)*
1. `DefaultLoginStrategyCacheService` wraps the three `GlobalState` keys (`CURRENT_LOGIN_STRATEGY_KEY`, `CACHE_KEY`, `CACHE_EXPIRATION_KEY`) previously inline in `LoginStrategyService`, making the cache a first-class collaborator that can be injected independently.
2. `clearCache()` atomically nulls all three state keys in sequence — makes it safe to call from the session-timeout handler without leaking partial state.

**`libs/auth/src/common/services/login-strategies/login-strategy-session-timeout.service.ts`** *(new)*
1. `DefaultLoginStrategySessionTimeoutService` registers the `loginStrategySessionTimeout` task handler in its constructor as a side-effect — this ensures the handler is always in place regardless of which DI context constructed the service.
2. The task handler sends `LOGIN_SESSION_EXPIRED` via `MessageSender` (cross-context broadcast) and then clears the cache; each step is wrapped in its own try/catch so a messaging failure does not prevent the cache from being cleared, and vice versa.
3. `loginSessionTimeout$` is derived from `MessageListener.messages$(LOGIN_SESSION_EXPIRED)` rather than a `BehaviorSubject`, so it works across extension background/popup boundaries without shared in-process state.
4. `startSessionTimeout()` cancels any in-flight timer before scheduling a new one, preventing double-fire on rapid re-authentication.

**`libs/auth/src/common/services/login-strategies/login-strategy.service.ts`**
1. Removed all three `GlobalState` field declarations (`currentAuthnTypeState`, `loginStrategyCacheState`, `loginStrategyCacheExpirationState`) and the `BehaviorSubject<boolean>` — replaced with injected `LoginStrategyCacheService` and `LoginStrategySessionTimeoutService` collaborators.
2. Removed `TaskSchedulerService` from the constructor; the task handler registration moved to `DefaultLoginStrategySessionTimeoutService` so it is decoupled from the login-flow orchestrator.
3. Removed `startSessionTimeout()` and `clearSessionTimeout()` private methods; delegated to `loginStrategySessionTimeoutService.startSessionTimeout()` and `.cancelSessionTimeout()`.
4. Removed `authenticationTimeoutSubject` and `authenticationSessionTimeout$` — the timeout signal is now broadcast via messaging and exposed through `LoginStrategySessionTimeoutService`.

**`libs/auth/src/common/services/index.ts`**
1. Added re-exports for `DefaultLoginStrategyCacheService` and `DefaultLoginStrategySessionTimeoutService` so DI registration files can import from the common barrel.

</details>

<details>
<summary>Components</summary>

**`libs/auth/src/angular/two-factor-auth/two-factor-auth.component.ts`**
1. Injected `LoginStrategySessionTimeoutService` and switched `listenForAuthnSessionTimeout()` to subscribe to `loginSessionTimeout$` instead of `loginStrategyService.authenticationSessionTimeout$` — removes the `if (!expired) return` guard since the new observable only emits on actual expiry (no boolean intermediary).

**`libs/auth/src/angular/new-device-verification/new-device-verification.component.ts`**
1. Same injection and subscription change as `TwoFactorAuthComponent` — the `loginStrategySessionTimeoutService.loginSessionTimeout$` observable replaces the boolean `authenticationSessionTimeout$` stream, simplifying the subscription body.

</details>

<details>
<summary>DI Registration</summary>

**`libs/angular/src/services/jslib-services.module.ts`**
1. Registered `LoginStrategyCacheService → DefaultLoginStrategyCacheService` with `[GlobalStateProvider]` deps, and `LoginStrategySessionTimeoutService → DefaultLoginStrategySessionTimeoutService` with `[TaskSchedulerService, LoginStrategyCacheService, LogService, MessageSender, MessageListener]` deps.
2. Updated `LoginStrategyServiceAbstraction` provider deps to replace `TaskSchedulerService` with `LoginStrategyCacheService` and `LoginStrategySessionTimeoutService`.

**`apps/browser/src/background/main.background.ts`**
1. Instantiated `DefaultLoginStrategySessionTimeoutService` directly in the background bootstrap (not bound to a token) purely for its constructor side-effect — ensures the `loginStrategySessionTimeout` task handler is registered in the extension background service worker before any login attempt.

**`apps/cli/src/service-container/service-container.ts`**
1. Registered `DefaultLoginStrategyCacheService` and `DefaultLoginStrategySessionTimeoutService` as local instances; passed `MessageListener.EMPTY` for the CLI since it has no cross-process messaging layer.
2. Updated `LoginStrategyService` instantiation to pass the two new service instances instead of `taskSchedulerService`.

</details>

<details>
<summary>Tests</summary>

**`libs/auth/src/common/services/login-strategies/login-strategy-cache.service.spec.ts`** *(new)*
1. Tests each write method (`setCurrentAuthType`, `setCacheData`, `setCacheExpiration`) verifying the underlying `GlobalState` is updated and the corresponding observable emits the new value.
2. Tests `clearCache()` verifying all three state keys are set to `null` after previously being populated.

**`libs/auth/src/common/services/login-strategies/login-strategy-session-timeout.service.spec.ts`** *(new)*
1. Tests that the task handler is registered in the constructor with the correct `ScheduledTaskNames.loginStrategySessionTimeout` key.
2. Tests that the handler sends `LOGIN_SESSION_EXPIRED` and calls `clearCache()`; independently tests that a `send()` failure is logged but does not block `clearCache()`, and that a `clearCache()` failure is logged.
3. Tests `loginSessionTimeout$` emits `void` when a `loginSessionExpired` message is received via `MessageListener`.
4. Tests `startSessionTimeout()` sets the cache expiration to `now + 5 minutes` using fake timers to eliminate date-assertion flakiness.
5. Tests `cancelSessionTimeout()` nulls the cache expiration and unsubscribes the active timer subscription.

**`libs/auth/src/common/services/login-strategies/login-strategy.service.spec.ts`**
1. Replaced `FakeGlobalState` + `TaskSchedulerService` mocks with mocks of `LoginStrategyCacheService` and `LoginStrategySessionTimeoutService` whose observables are backed by `BehaviorSubject`s wired to the mock method implementations — keeps test isolation without coupling the spec to internal state keys.
2. Updated the expired-session test to override `loginStrategyCacheService.cacheExpiration$` and `cacheData$` observables directly and re-instantiate `sut`, since the service no longer holds its own `GlobalState` references.
3. Added tests for the null-cache path (no cached session) and for non-API-error-triggered cache-clear paths to cover edge cases surfaced during refactor.

**`libs/auth/src/angular/two-factor-auth/two-factor-auth.component.spec.ts`**
1. Added `MockProxy<LoginStrategySessionTimeoutService>` with `loginSessionTimeout$` backed by a `BehaviorSubject<void>`, and provided it in the test module — required after the component no longer subscribes to `loginStrategyService.authenticationSessionTimeout$`.

</details>

## 📸 Recordings

<details>
<summary>Web</summary>

<details>
<summary>Password login works</summary>

https://github.com/user-attachments/assets/ff77c7ef-8b48-488b-8983-d2a3cfc3f692

</details>

<details>
<summary>Password login + new device verification works</summary>

https://github.com/user-attachments/assets/45f9e6db-88b1-4d04-a9bc-51a0ac71e53b

</details>

<details>
<summary>Password login + 2FA works</summary>

https://github.com/user-attachments/assets/60bad0e5-06b5-4a08-83ac-aaf1d50f0531

</details>

<details>
<summary>Login with device works</summary>

https://github.com/user-attachments/assets/90b4a2b3-f2dd-4f88-aeab-5360fc4be929

</details>

<details>
<summary>New device verification - authentication timeout works (temp set to 15 seconds)</summary>

https://github.com/user-attachments/assets/b10949fa-acd1-4fd1-98ac-668e248aa780

</details>

<details>
<summary>Login with passkey works</summary>

https://github.com/user-attachments/assets/862a577d-2682-417d-962b-df5d77096f23

</details>

<details>
<summary>2FA - authentication timeout works (temp set to 15 seconds)</summary>

https://github.com/user-attachments/assets/45f12337-3d80-4350-b949-a3064ab9d519

</details>

<details>
<summary>Login with SSO works</summary>

https://github.com/user-attachments/assets/2cf31e6d-8afb-4a9b-8cdb-30e63518017c

</details>

</details>

---

<details>
<summary>Desktop</summary>

<details>
<summary>Password login works</summary>

https://github.com/user-attachments/assets/4e78fa91-b562-472a-a010-b3f84b662656

</details>

<details>
<summary>Password login + new device verification works</summary>

https://github.com/user-attachments/assets/db36128f-d0b1-4867-b6aa-4fa327970cb3

</details>

<details>
<summary>Password login + 2FA works</summary>

https://github.com/user-attachments/assets/3347b8a6-8d7b-4482-87e3-8c022680a8e5

</details>

<details>
<summary>Login with device works</summary>

https://github.com/user-attachments/assets/af401889-2690-4de4-9611-44b17e5496cb

</details>

<details>
<summary>New device verification - authentication timeout works (temp set to 15 seconds)</summary>

https://github.com/user-attachments/assets/c0d47c67-ec5d-4d30-9e06-07cd0d429336

</details>

<details>
<summary>2FA - authentication timeout works (temp set to 15 seconds)</summary>

https://github.com/user-attachments/assets/edd1157b-a616-4edc-923c-1419ee24065a

</details>

<details>
<summary>Login with SSO works</summary>

https://github.com/user-attachments/assets/b7b475ea-f456-4941-94bc-829238ae803b

</details>

</details>

---

<details>
<summary>Extension</summary>

<details>
<summary>Password login + new device verification</summary>

https://github.com/user-attachments/assets/e10458e2-5bb3-47b5-b41b-890e20fde6fe

</details>

<details>
<summary>Password login + 2FA</summary>

https://github.com/user-attachments/assets/7b6c5d80-5270-4e36-b289-a5d684185ed2

</details>

<details>
<summary>Login with device works</summary>

https://github.com/user-attachments/assets/b3fe8746-850a-4bc2-abb9-0e2d4d5e722f

</details>

<details>
<summary>New device verification — auth timeout (15s) — popup persisted</summary>

https://github.com/user-attachments/assets/d840532e-cf31-4344-9ce2-5a52a95e5b40

</details>

<details>
<summary>New device verification — auth timeout (15s) — popup closed</summary>

https://github.com/user-attachments/assets/57fe6c41-f330-4b79-b2b9-8a3688aef8e3

</details>

<details>
<summary>2FA — auth timeout (15s) — popup persisted</summary>

https://github.com/user-attachments/assets/6d7add0b-017d-4b68-9606-66884d1ad9b6

</details>

<details>
<summary>2FA — auth timeout (15s) — popup closed</summary>

https://github.com/user-attachments/assets/bc1c0526-6d21-4f52-a7ac-66b5ea597aed

</details>

<details>
<summary>Login with SSO works</summary>

https://github.com/user-attachments/assets/5788afcd-2078-41a1-90d5-3b18213f9523

</details>

</details>

### CODE REVIEW

File: libs/auth/src/common/services/login-strategies/default-login-strategy-session-timeout.service.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    this.loginSessionTimeout$ = this.messageListener
+      .messages$(LOGIN_SESSION_EXPIRED)
+      .pipe(map((): void => undefined));
+
+    this.taskSchedulerService.registerTaskHandler(
+      ScheduledTaskNames.loginStrategySessionTimeout,
+      async () => {
+        try {
+          this.messageSender.send(LOGIN_SESSION_EXPIRED, {});
+          await this.loginStrategyCacheService.clearCache();
```
[the remark this answers is not in the document]
Comment (JaredSnider-Bitwarden) [reply libs/auth/src/common/services/login-strategies/default-login-strategy-session-timeout.service.ts:44]: Resolved with https://github.com/bitwarden/clients/pull/20108/commits/85d4881a4ca756ea1800de9785d04787b219a8f1

**Review by enmande — DISMISSED** [review]

**Review by enmande — APPROVED** [review]
